src/traces/table/plot.js

```diff
@@ -0,0 +1,892 @@
+}
+
+function columnMoved(gd, calcdata, i, indices) {
+    var o = calcdata[i][0].gdColumnsOriginalOrder;
+    calcdata[i][0].gdColumns.sort(function(a, b) {
+        return indices[o.indexOf(a)] - indices[o.indexOf(b)];
+    });
+
+    calcdata[i][0].columnorder = indices;
+
+    gd.emit('plotly_restyle');
+}
+
+function gridPick(spec, col, row) {
+    if(Array.isArray(spec)) {
+        var column = spec[Math.min(col, spec.length - 1)];
+        if(Array.isArray(column)) {
+            return column[Math.min(row, column.length - 1)];
+        } else {
+            return column;
+        }
+    } else {
+        return spec;
+    }
+}
+
+function easeColumn(selection, d, y) {
+    selection
+        .transition()
+        .ease(c.releaseTransitionEase, 1, 0.75)
+        .duration(c.releaseTransitionDuration)
+        .attr('transform', 'translate(' + d.x + ' ' + y + ')');
+}
+
+function cellsBlock(d) {return d.type === 'cells';}
+function headerBlock(d) {return d.type === 'header';}
+
+/**
+ * Revolver panel and cell contents layouting
+ */
+
+function splitToPanels(d) {
+    var prevPages = [0, 0];
+    var headerPanel = extendFlat({}, d, {
+        key: 'header',
+        type: 'header',
+        page: 0,
+        prevPages: prevPages,
+        currentRepaint: [null, null],
+        dragHandle: true,
+        values: d.calcdata.headerCells.values[d.specIndex],
+        rowBlocks: d.calcdata.headerRowBlocks,
+        calcdata: extendFlat({}, d.calcdata, {cells: d.calcdata.headerCells})
+    });
+    var revolverPanel1 = extendFlat({}, d, {
+        key: 'cells1',
+        type: 'cells',
+        page: 0,
+        prevPages: prevPages,
+        currentRepaint: [null, null],
+        dragHandle: false,
+        values: d.calcdata.cells.values[d.specIndex],
+        rowBlocks: d.calcdata.rowBlocks
+    });
+    var revolverPanel2 = extendFlat({}, d, {
+        key: 'cells2',
+        type: 'cells',
+        page: 1,
+        prevPages: prevPages,
+        currentRepaint: [null, null],
+        dragHandle: false,
+        values: d.calcdata.cells.values[d.specIndex],
+        rowBlocks: d.calcdata.rowBlocks
+    });
+    // order due to SVG using painter's algo:
+    return [revolverPanel1, revolverPanel2, headerPanel];
+}
+
+function splitToCells(d) {
+    var fromTo = rowFromTo(d);
+    return d.values.slice(fromTo[0], fromTo[1]).map(function(v, i) {
+        // By keeping identical key, a DOM node removal, creation and addition is spared, important when visible
+        // grid has a lot of elements (quadratic with xcol/ycol count).
+        // But it has to be busted when `svgUtil.convertToTspans` is used as it reshapes cell subtrees asynchronously,
+        // and by that time the user may have scrolled away, resulting in stale overwrites. The real solution will be
+        // to turn `svgUtil.convertToTspans` into a cancelable request, in which case no key busting is needed.
```

Comment:
> to turn `svgUtil.convertToTspans` into a cancelable request

Why not in this PR :smile: ?

Reply:
I'm meek 😺 Also, 'convertToTspans' creates temporary DOM nodes at the `document.body` level that get absorbed upon completion and there may be other lifecycle issues to wrap up, lest I add memory leaks

Reply:
Added this item too to https://github.com/plotly/plotly.js/issues/2053